common/djangoapps/student/models.py

```diff
@@ -1678,6 +1678,53 @@ def course_overview(self):
                 self._course_overview = None
         return self._course_overview
 
+    @property
+    def upgrade_deadline(self):
+        """
+        Returns the upgrade deadline for this enrollment, if it is upgradeable.
+
+        If the seat cannot be upgraded, None is returned.
+
+        Note:
+            When loading this model, use `select_related` to retrieve the associated schedule object.
+
+        Returns:
+            datetime|None
+        """
+        log.debug('Determining upgrade deadline for CourseEnrollment %d...', self.id)
+        if not CourseMode.is_mode_upgradeable(self.mode):
+            log.debug(
+                '%s mode of %s is not upgradeable. Returning None for upgrade deadline.', self.mode, self.course_id
+            )
```

Comment:
If you choose to keep these debug statements, can you prefix them with a common prefix (such as "Schedules: ") so it's easier for developers to filter them out, if needed (in case these become too noisy).